src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -1228,45 +1228,22 @@ public static void SerializeInner(object obj, BinaryTokenStreamWriter stream, Ty
 
         private static void WriteEnum(object obj, BinaryTokenStreamWriter stream, Type type)
         {
-            var underlyingType = Enum.GetUnderlyingType(type);
-            var size = Marshal.SizeOf(underlyingType);
-            switch (size)
-            {
-                case 1:
-                    stream.Write(Convert.ToByte(obj));
-                    break;
-                case 2:
-                    stream.Write(Convert.ToInt16(obj));
-                    break;
-                case 4:
-                    stream.Write(Convert.ToInt32(obj));
-                    break;
-                case 8:
-                    stream.Write(Convert.ToInt64(obj));
-                    break;
-                default:
-                    throw new NotSupportedException($"Serialization of type {type.GetParseableName()} is not supported.");
-            }
+            var t = Enum.GetUnderlyingType(type).TypeHandle;
+            if (t.Equals(byteTypeHandle) || t.Equals(sbyteTypeHandle)) stream.Write(Convert.ToByte(obj));
+            else if (t.Equals(shortTypeHandle) || t.Equals(ushortTypeHandle)) stream.Write(Convert.ToInt16(obj));
+            else if (t.Equals(intTypeHandle) || t.Equals(uintTypeHandle)) stream.Write(Convert.ToInt32(obj));
+            else if (t.Equals(longTypeHandle) || t.Equals(ulongTypeHandle)) stream.Write(Convert.ToInt64(obj));
+            else throw new NotSupportedException($"Serialization of type {type.GetParseableName()} is not supported.");
```

Comment:
what about 
        private static readonly RuntimeTypeHandle floatTypeHandle = typeof(float).TypeHandle;
        private static readonly RuntimeTypeHandle doubleTypeHandle = typeof(double).TypeHandle;
        private static readonly RuntimeTypeHandle charTypeHandle = typeof(char).TypeHandle;
        private static readonly RuntimeTypeHandle boolTypeHandle = typeof(bool).TypeHandle;

?
Do we serialize them?